src/MailCollector.php

```diff
@@ -1476,12 +1476,19 @@ public function getHeaders(\Laminas\Mail\Storage\Message $message)
             $subject = '';
         }
 
+       // secu on message_id setting
+        try {
+            $message_id = $message->getHeader('message_id')->getFieldValue();
+        } catch (Laminas\Mail\Storage\Exception\InvalidArgumentException $e) {
+            $message_id = '';
```

Comment:
I think it would be better to generate a unique value here since the message ID is used as an array key to uniquely identify the messages in a few places. Having a static prefix like "MISSING_ID_" followed by a random number perhaps?

Reply:
It looks like the Laminas library has a unique ID for each message in some cases, but it depends on the storage class/protocol used. For some, it defaults to the message ID.

A UUIDv4 would be sufficiently unique if we have to generate the ID.

At least in GLPI itself, the message ID isn't used to retrieve the same message again. I only see it used as an array key, matching followup emails to tickets (not an issue since GLPI sets the message ID), and stored in the database for not imported emails.

Reply:
Something like 
```$message_id = "MISSING_ID_".Toolbox::getRandomString(40);  ```
is easier to copy/paste... what do you think?